test/query.test.js

```diff
@@ -861,25 +861,63 @@ describe('query', () => {
     })
   })
 
-  // TODO Update this test to include `membership` Roles
-  test.skip('create_access_provider', async () => {
+  test('create_access_provider', async () => {
     const providerName = util.randomString('provider_')
     const issuerName = util.randomString('issuer_')
     const jwksUri = util.randomString()
     const fullUri = `https://${jwksUri}.auth0.com`
+    const roleOneName = util.randomString('role_one_')
+    const roleTwoName = util.randomString('role_two_')
 
-    // Successful instantiation with required fields
-    const provider = await adminClient.query(
-      query.CreateAccessProvider({
-        name: providerName,
-        issuer: issuerName,
-        jwks_uri: fullUri,
+    // Create roles
+    await adminClient.query(
+      query.CreateRole({
+        name: roleOneName,
+        privileges: [
+          {
+            resource: query.Databases(),
+            actions: { read: true },
+          },
+        ],
       })
     )
 
-    expect(provider.name).toEqual(providerName)
-    expect(provider.issuer).toEqual(issuerName)
-    expect(provider.jwks_uri).toEqual(fullUri)
+    await adminClient.query(
+      query.CreateRole({
+        name: roleTwoName,
+        privileges: [
+          {
+            resource: query.Databases(),
+            actions: { read: true },
+          },
+        ],
+      })
+    )
+
+    // Successful instantiation with required fields
+    try {
+      const provider = await adminClient.query(
+        query.CreateAccessProvider({
+          name: providerName,
+          issuer: issuerName,
+          jwks_uri: fullUri,
+          membership: [
+            query.Role(roleOneName),
+            {
+              role: query.Role(`${roleTwoName}`),
+              predicate: query.Query(query.Lambda(x => true)),
+            },
+          ],
+        })
+      )
+
+      expect(provider.name).toEqual(providerName)
+      expect(provider.issuer).toEqual(issuerName)
+      expect(provider.jwks_uri).toEqual(fullUri)
+      expect(provider.membership).toBeInstanceOf(Array)
+    } catch (error) {
+      expect(error).toBeInstanceOf(errors.BadRequest)
```

Comment:
Does this mean that if the try block fails, the test will still pass if the error is a BadRequest error? I imagine we'd want to require that the try block is successful.

Reply:
yeah, try/catch on tests might not be a good practice, you need to be sure your code will always succeed or fail and do the correct assertions.

Reply:
@marrony @nathaniel-may I'll split these tests in two so we can make assertions re: success and failure